pkg/testing/integration/program.go

```diff
@@ -1819,11 +1819,6 @@ func (pt *ProgramTester) copyTestToTemporaryDirectory() (string, string, error)
 	if copyErr := fsutil.CopyFile(tmpdir, sourceDir, nil); copyErr != nil {
 		return "", "", copyErr
 	}
-	// Reload the projinfo before making mutating changes (workspace.LoadProject caches the in-memory Project by path)
-	projinfo, err = pt.getProjinfo(projdir)
-	if err != nil {
-		return "", "", err
-	}
```

Comment:
I think the reload is necessary because the original projectinfo will refer to the file path for the source directory in testdata/whatever instead of tmpdir/whatever.

https://github.com/pulumi/pulumi/blob/b6795efbf16a363cfbf0da065f7fe083b0e74152/pkg/testing/integration/program.go#L1944

Reply:
Ah yes this is true because we're grabbing the dir as part of getProjInfo but the comment above this _is_ wrong. We don't cache by path, we just need the projinfo updated to the new dir.